### PR TITLE
Homework_8 Added JDBC and MySQL DB. With prepared Statements.

### PR DESCRIPTION
**Pull request by IgorPrihodko**

### CODE REVIEW

File: src/main/java/dao/basket/BasketDaoJDBC.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class BasketDaoJDBC implements BasketDao {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+    private static final Logger logger = Logger.getLogger(BasketDaoJDBC.class);
+    private static final String ADD_PRODUCT_TO_BASKET = "INSERT INTO list_of_products_in_basket " +
```
Comment (Sofasmile) [inline src/main/java/dao/basket/BasketDaoJDBC.java]: try to use more understandable name of table. for exmpl: basket_product

File: src/main/java/dao/basket/BasketDaoJDBC.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            "(basket_id, product_id) VALUES (?, ?)";
+    private static final String ADD_BASKET = "INSERT INTO baskets (user_id, total_price," +
```
Comment (Sofasmile) [inline src/main/java/dao/basket/BasketDaoJDBC.java]: it is bad practice to save total price

File: src/main/java/dao/basket/BasketDaoJDBC.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            " confirmation_code_id) VALUES (?, ?, ?)";
+    private static final String REMOVE_BASKET = "DELETE FROM baskets WHERE id = ?";
+    private static final String GET_LAST_BASKET_FOR_USER = "SELECT baskets.id, baskets.user_id," +
+            "baskets.total_price, baskets.confirmation_code_id FROM baskets INNER JOIN " +
```
Comment (Sofasmile) [inline src/main/java/dao/basket/BasketDaoJDBC.java]: do we need this inner join here?

File: src/main/java/utils/DBConnector.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.TimeZone;
+
+public class DBConnector {
+
+    private static final Logger logger = Logger.getLogger(DBConnector.class);
+    private static final String dbUrl = "jdbc:mysql://localhost:3306/online_shop?serverTimezone=" +
```
Comment (Sofasmile) [inline src/main/java/utils/DBConnector.java]: Hoe do you think, are this field: dbUrl, login, password constants?

**Review by Sofasmile — COMMENTED** [review]
good, but make small fixes

File: src/main/java/dao/basket/BasketDaoJDBC.java (hunk 1 above)
Comment (IgorPrihodko) [reply src/main/java/dao/basket/BasketDaoJDBC.java]: fixed

File: src/main/java/dao/basket/BasketDaoJDBC.java (hunk 3 above)
Comment (IgorPrihodko) [reply src/main/java/dao/basket/BasketDaoJDBC.java]: fixed

File: src/main/java/utils/DBConnector.java (hunk 4 above)
Comment (IgorPrihodko) [reply src/main/java/utils/DBConnector.java]: fixed. made them non-final.

File: src/main/java/dao/basket/BasketDaoJDBC.java (hunk 2 above)
Comment (IgorPrihodko) [reply src/main/java/dao/basket/BasketDaoJDBC.java]: are u talking just about DB, or about basket model too?

Comment (IgorPrihodko) [reply src/main/java/dao/basket/BasketDaoJDBC.java]: fixed

File: src/main/java/utils/DBConnector.java (hunk 4 above)
Comment (Sofasmile) [reply src/main/java/utils/DBConnector.java]: I think them can be constants, but then we need write name like this: DB_URL

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**